### ISSUE TITLE
psake on Linux

### ISSUE THREAD

**Issue by dermeister0** [psake/psake#262]
Here's a small instruction how to use psake on Linux.

- Install PowerShell Core

    https://docs.microsoft.com/en-us/powershell/scripting/setup/installing-powershell-core-on-linux?view=powershell-6

- Clone psake repository

    ```sh
    git clone https://github.com/psake/psake /opt/psake
    ```

- Create wrapper

    ```sh
    sudo sh -c 'echo "#!/bin/env sh\npwsh -Command \"& /opt/psake/src/psake.ps1 \$@; if (\\\$psake.build_success -eq \\\$false) { exit 1 } else { exit 0 }\"" > /usr/bin/psake'
    sudo chmod +x /usr/bin/psake
    ```

    It will create following script:

    ```sh
    #!/usr/bin/env sh
    pwsh -Command "& /opt/psake/src/psake.ps1 $@; if (\$psake.build_success -eq \$false) { exit 1 } else { exit 0 }"
    ```

I used Ansible to do the actions above. Tested on Ubuntu Server 18.04 and CentOS 7.

```yaml
- hosts: buildserver

  roles:
    - brentwg.powershell

  vars:
    ansible_become: true

  tasks:
    - name: Clone psake repository
      git:
        repo: 'https://github.com/psake/psake'
        dest: /opt/psake
    - name: Create psake wrapper
      copy:
        content: |
          #!/usr/bin/env sh
          pwsh -Command "& /opt/psake/src/psake.ps1 $@; if (\$psake.build_success -eq \$false) { exit 1 } else { exit 0 }"
        dest: /usr/bin/psake
        mode: 0755

  pre_tasks:
    - name: Enable Universe repository
      apt_repository:
        repo: deb http://archive.ubuntu.com/ubuntu bionic universe
      become: yes
      when: ansible_os_family == 'Debian'
```

**Comment by rkeithhill** [comment]
That's cool but perhaps we should put the shebang in `psake.ps1`, run `git update-index --chmod=+x src/psake.ps1` and commit the file so that it will be executable on Linux/macOS.

**Comment by dermeister0** [comment]
@rkeithhill
It's interesting that `pwsh psake.ps1` returns correct exit code, while `pwsh -File psake.ps1` always returns 0, as on Windows.

**Comment by rkeithhill** [comment]
You don't need a wrapper script but the script has to generate a **terminating** error for you to see a non-zero exit code.  This is how our PS build & test scripts work on Linux.  They report errors back to Jenkins by "throwing" errors (or converting non-terminating errors to terminating errors).  

Try this:
```
~$ cat test.ps1
#!/usr/bin/env pwsh
throw "kaboom!"

~$ chmod +x test.ps1

~$ ./test.ps1; echo $?
```
This outputs:
```
kaboom!
At /home/hillr/test.ps1:2 char:1
+ throw "kaboom!"
+ ~~~~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (kaboom!:String) [], RuntimeException
+ FullyQualifiedErrorId : kaboom!

1
```
To get your `Hi-There` missing command to error either wrap it in a try/catch (and rethrow the error) or use `$ErrorActionPreference = 'Stop'`.  

Also, if you care about running on macOS as well as linux then use this for a shebang:
```
#!/usr/bin/env pwsh
```
If you don't care about macOS, I would recommend this shebang:
```
#!/usr/bin/pwsh -noprofile
```

**Comment by dermeister0** [comment]
@rkeithhill 
I added following lines to the end of psake.ps1:

```powershell
if (!$psake.build_status) {
    exit 1
}
```

It works.

**Comment by devblackops** [comment]
Fixed in #267
